### PR TITLE
DBCON-222: Implement query-single operation.

### PR DESCRIPTION
**Pull request by emariotti3**

### CODE REVIEW

File: src/main/java/org/mule/extension/db/internal/result/row/NonStreamingInsensitiveMapRowHandler.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  public NonStreamingInsensitiveMapRowHandler(DbConnection dbConnection) {
+    super(dbConnection);
+  }
+
+  public NonStreamingInsensitiveMapRowHandler(DbConnection dbConnection, Charset charset) {
+    super(dbConnection, charset);
+  }
+
+
```
Comment (julodiaz) [inline src/main/java/org/mule/extension/db/internal/result/row/NonStreamingInsensitiveMapRowHandler.java]: Delete blank space.

File: src/main/java/org/mule/extension/db/internal/result/row/NonStreamingInsensitiveMapRowHandler.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  protected TypedValue<Object> handleSqlXmlType(SQLXML value) throws SQLException {
+    return new TypedValue<>(value.getString(), DataType.builder().type(SQLXML.class).mediaType(XML).build());
+  }
+
+  @Override
+  protected TypedValue<Object> handleBlobType(Blob value) throws SQLException {
+    ByteArrayInputStream is = new ByteArrayInputStream(IOUtils.toByteArray(value.getBinaryStream()));
+    return new TypedValue<>(IOUtils.toByteArray(is),
+                            DataType.builder().type(byte[].class).mediaType(BINARY).build());
+
```
Comment (julodiaz) [inline src/main/java/org/mule/extension/db/internal/result/row/NonStreamingInsensitiveMapRowHandler.java:46]: Delete blank space.

File: src/main/java/org/mule/extension/db/internal/result/row/NonStreamingInsensitiveMapRowHandler.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.SQLException;
+import java.sql.SQLXML;
+
+import static org.mule.runtime.api.metadata.MediaType.XML;
+import static org.mule.runtime.api.metadata.MediaType.BINARY;
+import static org.mule.runtime.api.metadata.MediaType.TEXT;
+
+public class NonStreamingInsensitiveMapRowHandler extends InsensitiveMapRowHandler {
```
Comment (julodiaz) [inline src/main/java/org/mule/extension/db/internal/result/row/NonStreamingInsensitiveMapRowHandler.java:26]: Add little comment/justification for the existence of this new class.

**Review by julodiaz — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/extension/db/internal/result/resultset/SingleResultSetHandler.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import org.mule.extension.db.internal.result.row.RowHandler;
+import org.mule.extension.db.internal.util.ResultSetCharsetEncodedHandler;
+import org.mule.runtime.core.api.util.CaseInsensitiveHashMap;
+
+import java.nio.charset.Charset;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Map;
+
+public class SingleResultSetHandler implements ResultSetCharsetEncodedHandler {
```
Comment (estebanwasinger) [inline src/main/java/org/mule/extension/db/internal/result/resultset/SingleResultSetHandler.java:19]: jdoc

File: src/main/java/org/mule/extension/db/internal/operation/DmlOperations.java [hunk 5]
```diff
@@ -158,6 +161,24 @@ public boolean useStickyConnections() {
     };
   }
 
+  @OutputResolver(output = SelectMetadataResolver.class)
+  public Map<String, Object> querySingle(@ParameterGroup(name = QUERY_GROUP) @Placement(
```
Comment (estebanwasinger) [inline src/main/java/org/mule/extension/db/internal/operation/DmlOperations.java:165]: jdoc

**Review by estebanwasinger — CHANGES_REQUESTED** [review]

**Review by julodiaz — APPROVED** [review]

File: src/main/java/org/mule/extension/db/internal/operation/DmlOperations.java (hunk 5 above)
Comment (estebanwasinger) [reply src/main/java/org/mule/extension/db/internal/operation/DmlOperations.java:178]: the Jdoc is also used for Mulesoft public docs try to give more love to it. ❤️

**Review by estebanwasinger — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/extension/db/internal/domain/metadata/QuerySingleMetadataResolver.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.extension.db.internal.domain.metadata;
+
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.metadata.MetadataContext;
+import org.mule.runtime.api.metadata.MetadataResolvingException;
+
+public class QuerySingleMetadataResolver extends SelectMetadataResolver {
```
Comment (estebanwasinger) [inline src/main/java/org/mule/extension/db/internal/domain/metadata/QuerySingleMetadataResolver.java:14]: jdoc

**Review by estebanwasinger — CHANGES_REQUESTED** [review]

**Review by estebanwasinger — APPROVED** [review]

**Review by julodiaz — APPROVED** [review]

**Review by nicomz — APPROVED** [review]

**Pull request merged**